src/aws-cur/cost-report.ts

```diff
@@ -0,0 +1,153 @@
+import { Stack, aws_cur, aws_iam, aws_s3 } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+
+/**
+ * Enum for the possible granularities of a cost report
+ */
+export class ReportGranularity {
+
+  /** Hourly granularity */
+  public static readonly HOURLY: ReportGranularity = new ReportGranularity('HOURLY');
+  /** Daily granularity */
+  public static readonly DAILY: ReportGranularity = new ReportGranularity('DAILY');
+  /** Weekly granularity */
+  public static readonly MONTHLY: ReportGranularity = new ReportGranularity('MONTHLY');
+
+  /**
+   * Returns a ReportGranularity instance for the given granularity string value.
+   *
+   * @param granularity - The granularity string value to create an instance for.
+   * @returns A ReportGranularity instance.
+   */
+  public static for(granularity: string): ReportGranularity {
+    return new ReportGranularity(granularity);
+  }
+
+  private constructor(public readonly value: string) { }
+}
+
+/**
+ * Enum for the possible formats of a cost report
+ */
+export class CurFormat {
+
+  /** GZIP compressed text or CSV format */
+  public static readonly TEXT_OR_CSV: CurFormat = new CurFormat('GZIP', 'textORcsv');
+  /** Parquet format */
+  public static readonly PARQUET: CurFormat = new CurFormat('Parquet', 'Parquet');
+
+  /**
+   * Returns a CurFormat instance for the given compression and format string values.
+   *
+   * @param compression - The compression string value
+   * @param format - The format string value
+   * @returns A CurFormat instance.
+   */
+  public static for(compression: string, format: string): CurFormat {
+    return new CurFormat(compression, format);
+  }
+
+  private constructor(public readonly compression: string, public readonly format: string) { }
```

Comment:
Someone may want to extend the class and create their own static variables instead of using factory methods. We would like to make it `protected` for this purpose. There would be no disadvantage to this.